allensdk/brain_observatory/behavior/data_objects/base/readable_interfaces.py

```diff
@@ -76,11 +77,27 @@ def from_nwb(cls, nwbfile: NWBFile) -> "DataObject":  # pragma: no cover
         raise NotImplementedError()
 
 
+class DataFileReadableInterface(abc.ABC):
+    """Maraks a data object as readable from various data files, not covered by
+    existing interfaces"""
+    @classmethod
+    @abc.abstractmethod
+    def from_data_file(cls, *args) -> "DataObject":
```

Comment:
Is there any way to be more specific than just *args? Ideally we would limit to only *kwargs since when inspecting method parameters `*args` is really uninformative.

Reply:
I agree with you. I don't think it's possible to be more specific here because it needs to accept arbitrary arguments. For example to instantiate an eye tracking object, it needs arguments specific to eye tracking. The one argument that we can be sure of is a `DataFile`, but oddly Pycharm complains if there is a single argument followed by `*args` that whatever arguments you have provided doesn't match the signature so I decided to leave it off. If you have suggestions for how to improve this by specifying concrete arguments let me know.